File: CoreJava/States-app/src/com/techlabs/states/StateService.java
package com.techlabs.states;

import java.io.*;
import java.util.*;
import java.util.Map.Entry;

public class StateService {
	Map<String, String> statelist = new HashMap<String, String>();

	public StateService() {
		init();
	}

	public void init() {
		try {
			FileReader fr = new FileReader("States.csv");
			BufferedReader br = new BufferedReader(fr);
			String list;
			while ((list = br.readLine()) != null) {
				String[] details = list.split(",");
				statelist.put(details[0], details[1]);
			}
			br.close();
		} catch (Exception e) {
			e.printStackTrace();
		}

	}

	public Map<String, String> search(String code) {
		code=code.toUpperCase();
		Map<String, String> temp = new HashMap<String, String>();
		for (Entry<String, String> state : statelist.entrySet()) {
			if (code.length() == 1
					&& state.getKey().startsWith(
							String.valueOf((code.charAt(0))))) {
				temp.put(state.getKey(), state.getValue());
			} else if (code.length() == 2 && statelist.containsKey(code)) {
				temp.put(code, statelist.get(code));
			} else if (state.getValue().contains(code)) {
				temp.put(code, state.getValue());
			}
		}
		return temp;
	}
}

File: DotNET/C#/StateServiceApp/StateServiceApp/StateService.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace StateServiceApp
{
    class StateService
    {
        Dictionary<String, String> states = new Dictionary<string, string>();

        public StateService()
        {
            init();
        }

        private void init()
        {
            String line;
            StreamReader streamReader = new StreamReader("States.csv");
            try
            {
                using (streamReader)
                {
                    while ((line = streamReader.ReadLine()) != null)
                    {
                        String[] state = line.Split(',');
                        states.Add(state[0], state[1]);
                    }
                }
            }
            catch
            {
                //Console.WriteLine("Exception occured");
            }
        }

        public Dictionary<String, String> Search(String code)
        {
            code = code.ToUpper();

            Dictionary<String, String> temp = new Dictionary<String, String>();

            foreach (KeyValuePair<string, string> entry in states)
            {

                if (code.Length == 1 && entry.Key.StartsWith(code.Substring(0, 1)))
                {
                    temp.Add(entry.Key, entry.Value);
                }
                else if (code.Length == 2 && states.ContainsKey(code))
                {
                    temp.Add(entry.Key, entry.Value);
                }
            }
            return temp;
        }
    }
}
